contracts/engine/GenArt721MinterDAExp_PBAB.sol

```diff
@@ -0,0 +1,611 @@
+        uint256 indexed projectId,
+        uint256 _auctionTimestampStart,
+        uint256 _priceDecayHalfLifeSeconds,
+        uint256 _startPrice,
+        uint256 _basePrice
+    );
+
+    /// Auction details cleared for project `projectId`.
+    event ResetAuctionDetails(uint256 indexed projectId);
+
+    /// Maximum and minimum allowed price decay half lifes updated.
+    event AuctionHalfLifeRangeSecondsUpdated(
+        uint256 _minimumPriceDecayHalfLifeSeconds,
+        uint256 _maximumPriceDecayHalfLifeSeconds
+    );
+
+    /// Core contract address this minter interacts with
+    address public immutable genArt721CoreAddress;
+
+    /// This contract handles cores with interface IGenArt721CoreV2_PBAB
+    IGenArt721CoreV2_PBAB private immutable genArtCoreContract;
+
+    uint256 constant ONE_MILLION = 1_000_000;
+
+    address payable public ownerAddress;
+    uint256 public ownerPercentage;
+
+    struct ProjectConfig {
+        bool maxHasBeenInvoked;
+        uint24 maxInvocations;
+        // max uint64 ~= 1.8e19 sec ~= 570 billion years
+        uint64 timestampStart;
+        uint64 priceDecayHalfLifeSeconds;
+        uint256 startPrice;
+        uint256 basePrice;
+    }
+
+    mapping(uint256 => ProjectConfig) public projectConfig;
+
+    /// Minimum price decay half life: price must decay with a half life of at
+    /// least this amount (must cut in half at least every N seconds).
+    uint256 public minimumPriceDecayHalfLifeSeconds = 300; // 5 minutes
+    /// Maximum price decay half life: price may decay with a half life of no
+    /// more than this amount (may cut in half at no more than every N seconds).
+    uint256 public maximumPriceDecayHalfLifeSeconds = 3600; // 60 minutes
+
+    // modifier to restrict access to only addresses specified by the
+    // `onlyWhitelisted()` method on the associated core contract
+    modifier onlyCoreAllowlisted() {
+        require(
+            genArtCoreContract.isWhitelisted(msg.sender),
+            "Only Core allowlisted"
+        );
+        _;
+    }
+
+    // modifier to restrict access to only the artist for a given project
+    modifier onlyArtist(uint256 _projectId) {
+        require(
+            (msg.sender ==
+                genArtCoreContract.projectIdToArtistAddress(_projectId)),
+            "Only Artist"
+        );
+        _;
+    }
+
+    // modifier to restrict access to calls only involving a valid projectId
+    // (an existing project)
+    modifier onlyValidProjectId(uint256 _projectId) {
+        require(
+            _projectId < genArtCoreContract.nextProjectId(),
+            "Only existing projects"
+        );
+        _;
+    }
+
+    /**
+     * @notice Initializes contract to be integrated with
+     * Art Blocks core contract at address `_genArt721Address`.
+     * @param _genArt721Address Art Blocks core contract address for
+     * which this contract will be a minter.
+     */
+    constructor(address _genArt721Address) ReentrancyGuard() {
```

Comment:
optional nit: guard preventing zero address as input parameter could be added

Reply:
Ack. Not doing this for now as this is a dev-x ask and we don't plan to deploy this minter for long.